File: src/main/java/composing/methods/OrderHelper.java
package composing.methods;

public class OrderHelper {
    private static final int MINIMUM_ITEMS_IN_STOCK = 10;

    public static boolean deserveDiscount(Order order) {
        return order.totalPrice() > 1000;
    }

    // Customer deserves a discount if customer age / number of products < 5
    public static boolean deserveDiscountBasedOnCustomer(Order order) {
        double nbOfProducts = order.getProducts().size();
        int customerAge = order.getCustomer().getAge();

        return (customerAge / nbOfProducts) < 5;
    }

    public static Stock calculateNewStock(Stock stock, int outFromStock) {
        int newStock = stock.getNbOfItems() - outFromStock;
        
        return new Stock(newStock < MINIMUM_ITEMS_IN_STOCK ?
                newStock + MINIMUM_ITEMS_IN_STOCK
                : newStock);
    }
}

File: src/scala/src/main/scala/composing/methods/OrderHelper.scala
package composing.methods

object OrderHelper {
  private val MinimumItemsInStock = 10

  def deserveDiscount(order: Order): Boolean = {
    val price = order.totalPrice
    price > 1000
  }

  // Customer deserves a discount if customer age / number of products < 5
  def deserveDiscountBasedOnCustomer(order: Order): Boolean = {
    val nbOfProducts = order.products.size
    val customerAge = order.customer.age

    val result = customerAge / nbOfProducts
    result < 5
  }

  def calculateNewStock(stock: Stock, outFromStock: Int): Int = {
    stock.nbOfItems = stock.nbOfItems - outFromStock
    if (stock.nbOfItems < MinimumItemsInStock)
      return stock.nbOfItems + MinimumItemsInStock
    stock.nbOfItems
  }
}
